service/worker/diagnostics/workflow.go

```diff
@@ -48,10 +49,25 @@ type DiagnosticsWorkflowInput struct {
 	RunID      string
 }
 
+type timeoutIssuesResult struct {
+	InvariantType    string
+	Reason           string
+	ExecutionTimeout *invariants.ExecutionTimeoutMetadata
+	ActivityTimeout  *invariants.ActivityTimeoutMetadata
+	ChildWfTimeout   *invariants.ChildWfTimeoutMetadata
+	DecisionTimeout  *invariants.DecisionTimeoutMetadata
+}
+
+type timeoutRootCauseResult struct {
+	RootCauseType        string
+	PollersMetadata      *invariants.PollersMetadata
+	HeartBeatingMetadata *invariants.HeartbeatingMetadata
+}
+
 type DiagnosticsWorkflowResult struct {
-	Issues    []invariants.InvariantCheckResult
-	RootCause []invariants.InvariantRootCauseResult
-	Runbooks  []string
+	TimeoutIssues    []*timeoutIssuesResult
```

Comment:
I think the result should have a grouped blocks:
{
   Problem1: { Issues: []Issue, Rootcause: "Rootcause", Runbook: "" }
   Problem2: ...
}

That will allow to group/dump it to the user in a more opinionated manner.

Reply:
good suggestion. updated the result